crypto/dsa/ecdsa/secp256k1.go

```diff
@@ -0,0 +1,84 @@
+package ecdsa
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+
+	_secp256k1 "github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4/ecdsa"
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+const (
+	SECP256K1JWA         string = "ES256K"
+	SECP256K1JWACurve    string = "secp256k1"
+	SECP256K1AlgorithmID string = SECP256K1JWACurve
+)
+
+func SECP256K1GeneratePrivateKey() (jwk.JWK, error) {
+	keyPair, err := _secp256k1.GeneratePrivateKey()
+	if err != nil {
+		return jwk.JWK{}, fmt.Errorf("failed to generate private key: %w", err)
+	}
+
+	dBytes := keyPair.Key.Bytes()
+	pubKey := keyPair.PubKey()
+	xBytes := pubKey.X().Bytes()
+	yBytes := pubKey.Y().Bytes()
+
+	privateKey := jwk.JWK{
+		KTY: KeyType,
+		CRV: SECP256K1JWACurve,
+		D:   base64.RawURLEncoding.EncodeToString(dBytes[:]),
+		X:   base64.RawURLEncoding.EncodeToString(xBytes),
+		Y:   base64.RawURLEncoding.EncodeToString(yBytes),
+	}
+
+	return privateKey, nil
+}
+
+func SECP256K1Sign(payload []byte, privateKey jwk.JWK) ([]byte, error) {
```

Comment:
It's a bit unfortunate that you couldn't implement or follow the standard library's crypto API, as it would make this library more generally useful in the Go ecosystem. It may be that there are slightly different requirements, but from the API I've seen that doesn't seem to be the case.

Reply:
the sole difference is using JWKs to represent private keys instead of `CryptoKey` or bytes.

the primary reason we chose to use `JWK` is because -- after implementing the original JS crypto package using a CryptoKey object that contained the key material as a byte array, We found ourselves having to convert to JWKs in almost every dependent library e.g. DIDs, VCs, for signing & encrypting with JWS/JWE, etc.

at every turn, the bytes format was having to be converted to and from JWK creating needless computation overhead and verbosity. the operative key format in nearly every standard we use ends up being JWK

so given that we're building on DIDs and VCs,   using JWK as our native key format makes life a lot easier

Another challenge we ran into with a byte array is that it doesn't carry the needed information with it so you end up having to cart around some metadata as well. additionally, unlike Go's CryptoKey, the web's CryptoKey or any other formats .... none of them is interoperable, whereas JWK is inherently interoperable no matter the programming language or runtime environment

Reply:
cc: @frankhinek

Reply:
Sounds reasonable. I just wanted to check that you were aware of it and had considered it, but if it doesn't make sense within the context of Web5 then SGTM.